x-pack/lib/geoip_database_management/extension.rb

```diff
@@ -4,18 +4,20 @@
 
 require "logstash/environment"
 
-module LogStash module Filters module Geoip
+module LogStash module GeoipDatabaseManagement
   class Extension < LogStash::UniversalPlugin
     include LogStash::Util::Loggable
 
     def additionals_settings(settings)
       require "logstash/runner"
       logger.trace("Registering additional geoip settings")
-      settings.register(LogStash::Setting::NullableString.new("xpack.geoip.download.endpoint"))
+      settings.register(LogStash::Setting::String.new("xpack.geoip.downloader.endpoint", "https://geoip.elastic.co/v1/database")
+                                                 .with_deprecated_alias("xpack.geoip.download.endpoint"))
+      settings.register(LogStash::Setting::TimeValue.new("xpack.geoip.downloader.poll.interval", "24h"))
```

Comment:
Aligns in both name and value-type with Elasticsearch's same feature. Intentionally not exposing it publicly because there is little value in doing so, and would likely require enhancing `TimeValue` with min/max bounds.